### PR TITLE
Update to Chromium 91.0.4472.77

### LINKED ISSUE #1522: Updating to Chromium 91
De facto  ... Google might jump to Chromium 91 on May 25  (2 days)  

https://www.chromestatus.com/features/schedule

Thats why i post this
Sorry for being annoying

### PR DESCRIPTION
**Pull request by Ahrotahn**
This PR updates to Chromium 91.0.4472.77.  

There is an upstream issue that causes a crash when closing the program.  This was solved in an update [here](https://chromium-review.googlesource.com/c/chromium/src/+/2821311), but didn't make it into this release.  So for the time being I've included the patch in `core/chromium-upstream/fix-crash-in-ThemeService.patch`.  

`use-local-devtools-files.patch` has been moved to `fix-building-with-prunned-binaries.patch` since it had been reduced to a single change that no longer had much to do with what the patch was originally made for.  The change removes a test that requires some binaries that are pruned.  So it could be removed entirely if the binaries under `third_party/devtools-frontend/src/test/e2e/resources/` are whitelisted instead.  

<br>

I've already completed a build on my end and haven't encountered any issues so far!  

<br>

[Blog Entry](https://chromereleases.googleblog.com/2021/05/stable-channel-update-for-desktop_25.html)

[Change Log](https://chromium.googlesource.com/chromium/src/+log/90.0.4430.212..91.0.4472.77?pretty=fuller&n=10000)

Fixes #1522

### CODE REVIEW

**Comment by PF4Public** [comment]
Did anyone get following error:
```
/base/clipboard/clipboard/clipboard_factory_ozone.o
In file included from ../../ui/base/clipboard/clipboard_factory_ozone.cc:11:
../../ui/base/clipboard/clipboard_non_backed.h:100:43: error: only virtual member functions can be marked 'override'
  bool IsSelectionBufferAvailable() const override;
                                          ^~~~~~~~
1 error generated.

```

**Comment by git-bruh** [comment]
https://github.com/chromium/chromium/commit/90544f821223a694e1af8adef621b8934201535a ?

**Comment by PF4Public** [comment]
@git-bruh Indeed! Looks like ozon could be enabled by default, I'll give it a try.

**Comment by lsfxz** [comment]
If it helps: I've been building it for quite some time now with

```
'use_ozone=true'
'ozone_platform_wayland=true'
'ozone_platform_x11=true'
'ozone_platform_headless=true'
'ozone_auto_platforms=false'
```

with no observable ill effects. If anyone wants to give it a spin, my most recent build (for Arch) is at https://gitlab.com/ohfp/caidao/-/jobs/1295950860/artifacts/file/caidao-91.0.4472.77-1-x86_64.pkg.tar.zst (caidao is pretty much just a rebranded ungoogled-chromium mostly for my own use, as it can coexist with regular chromium / ungoogled chromium – and includes `aarch64` builds ^^).

**Comment by git-bruh** [comment]
My personal builds are without VA-API but i know they changed some stuff about it in this release, so you need to pass `--enable-features=VaapiVideoDecoder` manually, this is probably what you need.

https://bugs.chromium.org/p/chromium/issues/detail?id=1207478

**Comment by PF4Public** [comment]
@git-bruh Thanks, that is exactly what I needed. Weird that they removed that flag, but didn't introduce another one for VA-API.

**Comment by Area51Kacz** [comment]
@lsfxz and what building platform are u using _(sometimes OBS turns kinda chaotic-problematic)_

**Comment by lsfxz** [comment]
After having looked "everywhere" for something to reliably build x86_64 and aarch64 packages on (and with enough "free minutes" etc. for things like chrome)… in the end I've just resorted to directly using gitlab runners + docker on a VPS (x86_64) or on an aarch64 instance I spin up whenever I build something large (like this ^^).
Free would be nice, but for things like chromium on aarch64, it's just not fun any other way (maybe qemu on x86_64 might help, but at some point I just wanted things to work ^^)

**Comment by Area51Kacz** [comment]
And does someone knows why this PR gaves me this weird bug

`patching file google_apis/google_api_keys.cc
patching file sandbox/linux/seccomp-bpf-helpers/baseline_policy.cc
Hunk #1 succeeded at 286 with fuzz 1 (offset 29 lines).
patching file sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc
patching file sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.h
patching file sandbox/linux/services/syscall_wrappers.cc
patching file sandbox/linux/services/syscall_wrappers.h
patching file base/check.h
patching file build/linux/unbundle/icu.gn
patching file sql/recover_module/btree.cc
patching file sql/recover_module/btree.h
patching file sql/recover_module/cursor.cc
patching file sql/recover_module/cursor.h
patching file sql/recover_module/pager.cc
patching file sql/recover_module/pager.h
patching file chrome/browser/flag-metadata.json
patching file ui/gl/gl_image_native_pixmap.cc
Hunk #1 succeeded at 289 (offset 1 line).
patching file gpu/command_buffer/service/error_state.cc
  -> Pruning binaries
  -> Applying patches
INFO: Applying patches from /home/senpai/funstuff/ungoogled-chromium/src/ungoogled-chromium-91.0.4472.77-1/patches
INFO: * Applying 0001-fix-building-without-safebrowsing.patch (1/91)
patching file chrome/browser/BUILD.gn
patching file chrome/browser/browser_process.h
patching file chrome/browser/browser_process_impl.cc
patching file chrome/browser/browser_process_impl.h
patching file chrome/browser/browsing_data/chrome_browsing_data_remover_delegate.cc
patching file chrome/browser/chrome_content_browser_client.cc
patching file chrome/browser/component_updater/file_type_policies_component_installer.cc
patching file chrome/browser/download/chrome_download_manager_delegate.cc
patching file chrome/browser/download/download_item_model.cc
patching file chrome/browser/download/download_prefs.cc
patching file chrome/browser/download/download_target_determiner.cc
patching file chrome/browser/extensions/api/downloads/downloads_api.cc
patching file chrome/browser/extensions/api/downloads/downloads_api.h
patching file chrome/browser/extensions/api/safe_browsing_private/safe_browsing_private_api.cc
patching file chrome/browser/extensions/api/webstore_private/webstore_private_api.cc
patching file chrome/browser/extensions/blocklist_state_fetcher.cc
patching file chrome/browser/extensions/blocklist_state_fetcher.h
patching file chrome/browser/metrics/chrome_metrics_service_client.cc
patching file chrome/browser/net/trial_comparison_cert_verifier_controller.cc
patching file chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc
patching file chrome/browser/profiles/profile_impl.cc
patching file chrome/browser/profiles/profiles_state.cc
patching file chrome/browser/safe_browsing/BUILD.gn
patching file chrome/browser/ssl/security_state_tab_helper.cc
patching file chrome/browser/subresource_filter/chrome_subresource_filter_client.cc
patching file chrome/browser/ui/BUILD.gn
patching file chrome/browser/ui/tab_helpers.cc
patching file chrome/browser/ui/webui/downloads/downloads_dom_handler.cc
patching file chrome/browser/ui/webui/downloads/downloads_dom_handler.h
patching file chrome/browser/ui/webui/interstitials/interstitial_ui.cc
patching file chrome/renderer/chrome_content_renderer_client.cc
patching file components/security_interstitials/content/cert_report_helper.cc
INFO: * Applying 0003-disable-autofill-download-manager.patch (2/91)
patching file components/autofill/core/browser/autofill_download_manager.cc
INFO: * Applying 0005-disable-default-extensions.patch (3/91)
patching file chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
patching file chrome/browser/extensions/component_loader.cc
patching file chrome/browser/extensions/external_component_loader.cc
patching file chrome/browser/extensions/webstore_installer.cc
INFO: * Applying 0007-disable-web-resource-service.patch (4/91)
patching file components/web_resource/web_resource_service.cc
INFO: * Applying 0009-disable-google-ipv6-probes.patch (5/91)
patching file net/dns/host_resolver_manager.cc
INFO: * Applying 0015-disable-update-pings.patch (6/91)
patching file chrome/updater/configurator.cc
INFO: * Applying 0017-disable-new-avatar-menu.patch (7/91)
patching file components/signin/internal/identity_manager/primary_account_policy_manager_impl.cc
INFO: * Applying 0021-disable-rlz.patch (8/91)
patching file BUILD.gn
patching file rlz/buildflags/buildflags.gni
INFO: * Applying unrar.patch (9/91)
patching file chrome/browser/safe_browsing/download_protection/file_analyzer.cc
patching file chrome/common/safe_browsing/BUILD.gn
patching file chrome/services/file_util/safe_archive_analyzer.cc
INFO: * Applying safe_browsing-disable-incident-reporting.patch (10/91)
patching file chrome/browser/safe_browsing/incident_reporting/incident_report_uploader_impl.cc
patching file chrome/browser/safe_browsing/incident_reporting/incident_reporting_service.cc
patching file chrome/browser/safe_browsing/safe_browsing_blocking_page.cc
patching file chrome/browser/safe_browsing/safe_browsing_service.cc
patching file components/security_interstitials/core/safe_browsing_loud_error_ui.cc
INFO: * Applying safe_browsing-disable-reporting-of-safebrowsing-over.patch (11/91)
patching file components/safe_browsing/content/browser/client_side_detection_service.cc
INFO: * Applying all-add-trk-prefixes-to-possibly-evil-connections.patch (12/91)
patching file build/apple/tweak_info_plist.py
patching file chrome/browser/chromeos/customization/customization_document.cc
patching file chrome/browser/chromeos/extensions/file_manager/private_api_drive.cc
patching file chrome/browser/chromeos/extensions/file_manager/private_api_misc.cc
patching file chrome/browser/chromeos/policy/remote_commands/crd_host_delegate.cc
patching file chrome/browser/extensions/api/cryptotoken_private/cryptotoken_private_api.cc
patching file chrome/browser/extensions/install_signer.cc
patching file chrome/browser/media/webrtc/webrtc_event_log_uploader.cc
patching file chrome/browser/media/webrtc/webrtc_log_uploader.cc
patching file chrome/browser/nacl_host/nacl_infobar_delegate.cc
patching file chrome/browser/profiles/profile_avatar_downloader.cc
patching file chrome/browser/resources/default_apps/external_extensions.json
patching file chrome/browser/safe_browsing/download_protection/download_feedback.cc
patching file chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc
patching file chrome/browser/supervised_user/supervised_user_service.cc
patching file chrome/browser/tracing/crash_service_uploader.cc
patching file chrome/browser/ui/views/outdated_upgrade_bubble_view.cc
patching file chrome/browser/ui/webui/ntp/ntp_resource_cache.cc
patching file chrome/chrome_cleaner/components/recovery_component.cc
patching file chrome/chrome_cleaner/crash/crashpad_crash_reporter.cc
patching file chrome/common/extensions/chrome_extensions_client.cc
patching file chrome/installer/setup/google_chrome_behaviors.cc
patching file chromecast/browser/service/cast_service_simple.cc
patching file chromecast/crash/linux/minidump_uploader.cc
patching file chromeos/geolocation/simple_geolocation_provider.cc
patching file components/cloud_devices/common/cloud_devices_urls.cc
patching file components/drive/service/drive_api_service.cc
patching file components/feedback/feedback_uploader.cc
patching file components/google/core/common/google_util.cc
patching file components/history/core/browser/web_history_service.cc
patching file components/metrics/url_constants.cc
patching file components/password_manager/core/browser/password_store.cc
patching file components/safe_browsing/content/browser/client_side_detection_service.cc
patching file components/safe_search_api/safe_search/safe_search_url_checker_client.cc
patching file components/safe_search_api/stub_url_checker.cc
patching file components/translate/core/browser/translate_url_fetcher.cc
patching file components/translate/core/common/translate_util.cc
patching file components/variations/variations_url_constants.cc
patching file content/browser/speech/speech_recognition_engine.cc
patching file content/browser/webauth/authenticator_common.cc
patching file content/shell/browser/shell_browser_main_parts.cc
patching file extensions/common/extension_urls.cc
patching file google_apis/gaia/gaia_constants.cc
patching file google_apis/gaia/gaia_urls.cc
patching file google_apis/gcm/engine/gservices_settings.cc
patching file jingle/notifier/base/gaia_token_pre_xmpp_auth.cc
patching file remoting/base/breakpad_mac.mm
patching file remoting/protocol/jingle_messages.cc
patching file rlz/lib/lib_values.cc
patching file third_party/libjingle_xmpp/xmpp/constants.cc
patching file ui/accessibility/extensions/chromevoxclassic/chromevox/background/prefs.js
patching file ui/accessibility/extensions/chromevoxclassic/host/chrome/host.js
patching file ui/views/examples/webview_example.cc
INFO: * Applying disable-crash-reporter.patch (13/91)
patching file chrome/browser/tracing/crash_service_uploader.cc
INFO: * Applying disable-google-host-detection.patch (14/91)
patching file chrome/common/google_url_loader_throttle.cc
patching file components/google/core/common/google_util.cc
patching file components/page_load_metrics/browser/page_load_metrics_util.cc
patching file components/page_load_metrics/common/page_load_metrics_util.cc
patching file components/search_engines/template_url.cc
patching file components/variations/net/variations_http_headers.cc
patching file net/base/url_util.cc
INFO: * Applying replace-google-search-engine-with-nosearch.patch (15/91)
patching file components/search_engines/prepopulated_engines.json
INFO: * Applying disable-signin.patch (16/91)
patching file chrome/browser/ui/chrome_pages.cc
INFO: * Applying toggle-translation-via-switch.patch (17/91)
patching file chrome/browser/renderer_context_menu/render_view_context_menu.cc
patching file components/translate/core/browser/translate_language_list.cc
patching file components/translate/core/browser/translate_manager.cc
patching file components/translate/core/browser/translate_ranker_impl.cc
patching file components/translate/core/browser/translate_script.cc
INFO: * Applying disable-untraceable-urls.patch (18/91)
patching file chrome/browser/plugins/plugins_resource_service.cc
patching file components/safe_browsing/content/browser/client_side_model_loader.cc
patching file components/safe_browsing/content/browser/client_side_model_loader.h
patching file rlz/lib/financial_ping.cc
patching file rlz/lib/lib_values.cc
patching file rlz/lib/lib_values.h
INFO: * Applying disable-profile-avatar-downloading.patch (19/91)
patching file chrome/browser/profiles/profile_avatar_downloader.cc
INFO: * Applying disable-gcm.patch (20/91)
patching file components/gcm_driver/gcm_client_impl.cc
INFO: * Applying disable-domain-reliability.patch (21/91)
patching file chrome/browser/domain_reliability/service_factory.cc
patching file components/domain_reliability/BUILD.gn
patching file components/domain_reliability/bake_in_configs.py
patching file components/domain_reliability/google_configs.cc
patching file components/domain_reliability/uploader.cc
INFO: * Applying block-trk-and-subdomains.patch (22/91)
patching file chrome/browser/autocomplete/chrome_autocomplete_scheme_classifier.cc
patching file chrome/browser/history/history_utils.cc
patching file chrome/browser/ui/singleton_tabs.cc
patching file components/omnibox/browser/autocomplete_input.cc
patching file components/url_formatter/url_fixer.cc
patching file content/browser/child_process_security_policy_impl.cc
patching file net/BUILD.gn
The next patch would create the file net/url_request/trk_protocol_handler.cc,
which already exists!  Skipping patch.
1 out of 1 hunk ignored
The next patch would create the file net/url_request/trk_protocol_handler.h,
which already exists!  Skipping patch.
1 out of 1 hunk ignored
patching file net/url_request/url_request.cc
patching file net/url_request/url_request_context_builder.cc
patching file url/url_constants.cc
patching file url/url_constants.h
patching file url/url_util.cc
Traceback (most recent call last):
  File "/home/senpai/funstuff/ungoogled-chromium/src/ungoogled-chromium-91.0.4472.77-1/utils/patches.py", line 250, in <module>
    main()
  File "/home/senpai/funstuff/ungoogled-chromium/src/ungoogled-chromium-91.0.4472.77-1/utils/patches.py", line 246, in main
    args.callback(args, parser.error)
  File "/home/senpai/funstuff/ungoogled-chromium/src/ungoogled-chromium-91.0.4472.77-1/utils/patches.py", line 198, in _apply_callback
    apply_patches(
  File "/home/senpai/funstuff/ungoogled-chromium/src/ungoogled-chromium-91.0.4472.77-1/utils/patches.py", line 132, in apply_patches
    subprocess.run(cmd, check=True)
  File "/usr/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/patch', '-p1', '--ignore-whitespace', '-i', '/home/senpai/funstuff/ungoogled-chromium/src/ungoogled-chromium-91.0.4472.77-1/patches/core/ungoogled-chromium/block-trk-and-subdomains.patch', '-d', '.', '--no-backup-if-mismatch', '--forward']' returned non-zero exit status 1.
==> ERROR: A failure occurred in prepare().
    Aborting...
`

**Comment by Ahrotahn** [comment]
@Area51Kacz That error means you're building with a version of `gn` from before September's changes, updating it to a more recent version should allow you to continue.

**Comment by Area51Kacz** [comment]
@Ahrotahn @lsfxz and .. should i remove vaapi falgs

and replace them with ozone flags  or both could coexist

**Comment by lsfxz** [comment]
1: I've never gotten it to work on my machine ever reliably anyway (neither on wayland nor on x11). Still: you can build it with vaapi and ozone flags, but you cannot use them both when running it – if I use the vaapi flag, it runs under xwayland and not wayland-natively. Having native wayland support and hw video acceleration… seems to only work on Firefox, for the time being.

2: no idea

3: Upstream Arch uses this patch: https://github.com/stha09/chromium-patches/blob/master/chromium-90-ruy-include.patch (https://github.com/archlinux/svntogit-packages/blob/packages/chromium/trunk/PKGBUILD#L114). Not sure if that's included in the ungoogled-archlinux repo / PKGBUILD yet, but having it in my own PKGBUILDs makes things work (for my builds, at least ^^)

**Comment by Area51Kacz** [comment]
so actually    i could use BOTH (vaapi and ozone flags)  without any majoral drama  (on build process)   and run  it     as normal

File: patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch [hunk 1]
```diff
@@ -184,198 +184,241 @@
...
  std::vector<std::string> ContextInfoFetcher::GetOnSecurityEventProviders() {
    return connectors_service_->GetReportingServiceProviderNames(
        enterprise_connectors::ReportingConnector::SECURITY_EVENT);
+--- a/chrome/browser/enterprise/signals/context_info_fetcher.h
++++ b/chrome/browser/enterprise/signals/context_info_fetcher.h
+@@ -6,10 +6,10 @@
+ #define CHROME_BROWSER_ENTERPRISE_SIGNALS_CONTEXT_INFO_FETCHER_H_
+ 
+ #include <string>
++#include <memory>
```
Comment (Eloston) [inline patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch:215]: Why is this include needed?

**Review by Eloston — COMMENTED** [review]
LGTM, just one small question

**Comment by lsfxz** [comment]
> so actually i could use BOTH (vaapi and ozone flags) without any majoral drama (on build process) and run it as normal

Yep. At least that seems to work for me ^^

File: patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch (hunk 1 above)
Comment (Ahrotahn) [reply patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch:215]: `std::unique_ptr` is used around line 57 which was previously getting access to it from one of `safe_browsing_prefs.h`'s includes.

**Pull request merged**